Expensify/App
# Save preferred locale in NVP

cc @chiragsalian 

Related Web-E PR: https://github.com/Expensify/Web-Expensify/pull/31508

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/167702

### Tests
1. Open app in two different devices, e.g. web and desktop and login with the same user.
2. On web, go to `Settings > Preferences` and change the `Language`.
3. Confirm that the language changes on Desktop too and vice-versa.

### QA Steps
Steps above.

### Tested On


### Screenshots

#### Web

https://user-images.githubusercontent.com/22219519/126234866-a21cff2c-d4c4-451b-b67e-e1563325312a.mov

#### Mobile Web

https://user-images.githubusercontent.com/22219519/126234878-b8bb586c-506f-49de-9350-091dbd6cd735.mov

#### Desktop

https://user-images.githubusercontent.com/22219519/126234935-1f40f4c9-3bb5-48b3-a34e-12099b96f5c1.mov

#### iOS

https://user-images.githubusercontent.com/22219519/126234893-a1a0b70b-b84c-4f0b-bf01-207fcf1beab7.mov

#### Android

https://user-images.githubusercontent.com/22219519/126234954-7d9f2e7e-3e15-4952-8d2b-8da72194e6fd.mov
